pkg/controller/signer/signer-controller.go

```diff
@@ -251,7 +254,7 @@ func signerFailure(r *ReconcileCSR, csr *csrv1.CertificateSigningRequest, reason
 
 // Update the status conditions on the CSR object
 func updateCSRStatusConditions(r *ReconcileCSR, csr *csrv1.CertificateSigningRequest, reason string, message string) {
-	csr.Status.Conditions = append(csr.Status.Conditions, csrv1.CertificateSigningRequestCondition{
+	setCertificateSigningRequestCondition(&csr.Status.Conditions, csrv1.CertificateSigningRequestCondition{
```

Comment:
Why not use this? https://github.com/openshift/cluster-network-operator/blob/621fb760a947a29ebe92dce7188526f5a7bbbe56/vendor/k8s.io/apimachinery/pkg/api/meta/conditions.go#L31

Reply:
this is updating `k8s.io/api/certificates/v1.CertificateSigningRequestCondition` object here, so not using generic Condition object.